pkg/loadbalancerservice/router.go

```diff
@@ -0,0 +1,135 @@
+package loadbalancerservice
+
+import (
+	"slices"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultRouterServiceName            = "router-default"
+	defaultRouterServiceNamespace       = "openshift-ingress"
+	defaultRouterServiceAnnotationKey   = "ingresscontroller.operator.openshift.io/owning-ingresscontroller"
+	defaultRouterServiceAnnotationValue = "default"
+)
+
+type serviceUpdateFunction func([]string) error
+
+func defaultRouterWatch(ipAddresses, nicNames []string, updateFunc serviceUpdateFunction, stopCh <-chan struct{}) {
+	for {
+		ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+		if err != nil {
+			klog.Errorf("unable to determine default router listening addresses: %v", err)
+			continue
+		}
+		if err := updateFunc(ips); err != nil {
+			klog.Errorf("unable to update default router service status: %v", err)
+			continue
+		}
+		break
+	}
+
+	updateChan := make(chan netlink.AddrUpdate)
+	doneChan := make(chan struct{})
+	err := netlink.AddrSubscribe(updateChan, doneChan)
+	if err != nil {
+		klog.Errorf("unable to subscribe to IP address changes: %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-updateChan:
+			ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+			if err != nil {
+				klog.Errorf("unable to determine default router listening addresses: %v", err)
+				break
+			}
+			err = updateFunc(ips)
+			klog.Infof("P333. This is something I am waiting for: %v", err)
+			if err != nil {
+				klog.Errorf("unable to update default router service status: %v", err)
+				break
```

Comment:
Why not retry in some time? I guess this also leaves MicroShift without this "controller"?

Reply:
Will include this as part of the initial loop. However, falling into this error would mean no more dynamically changing the load balancer IP addresses. The last good known state would remain until an admin took action by restarting MicroShift. This may be considered as a degraded state, but not failed.

Reply:
Exactly, but are we communicating this in any other way than logging it? How does the admin should recognize microshift needs restarting?

Reply:
There is no other way right now but logs and docs. This situation shall be explicitly stated there as a degraded condition alongside a remediation procedure to solve it.

Reply:
ack